test/unit/TempVolume.test.ts

```diff
@@ -1,25 +1,125 @@
+import { promises as fs, existsSync } from 'fs'
+import { resolve } from 'path'
+import {
+	safeResolve,
+	createTempDir,
+	destroyTempDir,
+	withTempDir,
+} from '../../src/Daemon/TempVolume'
+
 describe('safeResolve', () => {
-	it.todo('correctly resolves safe paths')
-	it.todo('rejects relative paths that escape the root')
-	it.todo('rejects absolute paths that escape the root')
-	it.todo('rejects paths will null bytes')
-	it.todo('rejects NTFS special paths')
+	it('correctly resolves safe paths', () => {
+		// test absolute path only
+		const path1 = safeResolve('/usr/bin/ls', '')
+		expect(path1).toEqual('/usr/bin/ls')
+
+		// test absolute path with file
+		const path2 = safeResolve('/usr/bin', 'ls')
+		expect(path2).toEqual('/usr/bin/ls')
+	})
+	it('rejects relative paths that escape the root', () => {
+		const resolved = safeResolve('../', 'ls')
+		expect(resolved).toBeUndefined()
+	})
+	it('rejects absolute paths that escape the root', () => {
+		const resolved = safeResolve('/../', 'ls')
+		expect(resolved).toBeUndefined()
+	})
```

Comment:
`safeResolve` needs the first parameter (`root`) to be an absolute path, but you're testing relative paths here. The cases we actually want to prevent are like this:
```ts
safeResolve('/bin/safe', '../risky')
safeResolve('/bin/safe', '/sbin/su')
```